pkg/controllers/clusterexternalsecret/clusterexternalsecret_controller.go

```diff
@@ -0,0 +1,185 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterexternalsecret
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	esv1alpha1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1alpha1"
+)
+
+// ClusterExternalSecretReconciler reconciles a ClusterExternalSecret object.
+type Reconciler struct {
+	client.Client
+	Log             logr.Logger
+	Scheme          *runtime.Scheme
+	RequeueInterval time.Duration
+}
+
+const (
+	errGetCES              = "could not get ClusterExternalSecret"
+	errPatchStatus         = "unable to patch status"
+	errLabelMap            = "unable to get map from labels"
+	errNamespaces          = "could not get namespaces from selector"
+	errGetExistingES       = "could not get existing ExternalSecret"
+	errCreatingOrUpdating  = "could not create or update ExternalSecret"
+	errSetCtrlReference    = "could not set the controller owner reference"
+	errSecretAlreadyExists = "secret already exists in namespace"
+)
+
+//+kubebuilder:rbac:groups=external-secrets.io,resources=clusterexternalsecrets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=external-secrets.io,resources=clusterexternalsecrets/status,verbs=get;update;patch
+
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("ClusterExternalSecret", req.NamespacedName)
+
+	var clusterExternalSecret esv1alpha1.ClusterExternalSecret
+
+	err := r.Get(ctx, req.NamespacedName, &clusterExternalSecret)
+	if apierrors.IsNotFound(err) {
+		return ctrl.Result{}, nil
+	} else if err != nil {
+		log.Error(err, errGetCES)
+		return ctrl.Result{}, nil
+	}
+
+	p := client.MergeFrom(clusterExternalSecret.DeepCopy())
+	defer func() {
+		err = r.Status().Patch(ctx, &clusterExternalSecret, p)
+		if err != nil {
+			log.Error(err, errPatchStatus)
+		}
+	}()
+
+	// Fetch Namespaces to grab ExternalSecrets
+	genClient := kubernetes.NewForConfigOrDie(ctrl.GetConfigOrDie())
+	namespaces := genClient.CoreV1().Namespaces()
+
+	refreshInt := r.RequeueInterval
+	if clusterExternalSecret.Spec.RefreshInterval != nil {
+		refreshInt = clusterExternalSecret.Spec.RefreshInterval.Duration
+	}
+
+	labelMap, err := metav1.LabelSelectorAsMap(&clusterExternalSecret.Spec.NamespaceSelector)
+	if err != nil {
+		log.Error(err, errLabelMap)
+		return ctrl.Result{RequeueAfter: refreshInt}, nil
```

Comment:
Since this is an error condition that we want to keep trying to reconcile, I suppose the best approach here would be to export the error
```suggestion
		return ctrl.Result{RequeueAfter: refreshInt}, err
```